samples/kotlin-android-with-moshi-kotlin-codegen/build.gradle.kts

```diff
@@ -0,0 +1,56 @@
+plugins {
```

Comment:
Do we really need a new sample? We're switching the whole stack from Reflect to Codegen. The changes will be already reflected on the `kotlin-android`, `junit-tests`, `generated-code` samples

Reply:
The idea of this PR was mostly to allow both to work.

Ideally I'd like to have a version released that supports both approaches and then if we decide that generated-code is the way forward (the performance improvements are balanced by extra size, etc. ) than we'll advocate only for generated-code removing the reflection approach from the codebase

Reply:
We did chat around this specific issue offline and we decided that:
 * for the next minor releases the plugin will support both reflective and codegen approaches
 * for the next major release we will be support only codegen approach by still preserving the capability of injecting `KotlinJsonAdapterFactory` (reflective approach).

We've decided that with this approach we can limit eventual issues that might surface by proguard configurations as well as gaining confidence around the final approach.

I'll be creating an issue (that will act as reminder) to fully remove the reflective approach